lib/game.rb

```diff
@@ -0,0 +1,137 @@
+require 'pry'
+
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  # Define your WIN_COMBINATIONS constant
+  WIN_COMBINATIONS = [
+    [0,1,2], # Top row
+    [3,4,5], # Middle row
+    [6,7,8], # Bottom row
+    [0,3,6], # Left column
+    [1,4,7], # Middle column
+    [2,5,8], # Right column
+    [0,4,8], # First diagonal
+    [2,4,6]  # Second diagonal
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def turn_count
+    board.cells.count{|token| token == "X" || token == "O"}
+  end
+
+  def current_player
+    turn_count % 2 == 0 ? player_1 : player_2
+    #if even, return player_1
+  end
+
+  def over?
+    #game is over if board is full or someone won
+    if draw? || won?
+      true
+    else
+      false
+    end
+  end
+
+  def draw?
+    !won? && full?
+  end
+
+  def full?
+    board.cells.all? {|p| p == "X" || p == "O"}
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |win_set|
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```